semmc-ppc/src/SemMC/Architecture/PPC32.hs

```diff
@@ -155,12 +156,14 @@ instance T.TemplatableOperand PPC where
             where mkTemplate gprNum gprOffset =
                       T.TemplatedOperand Nothing (Set.fromList [Some (LocGPR (PPC.GPR gprNum)),Some (LocGPR (PPC.GPR gprOffset))]) mkTemplate' :: T.TemplatedOperand PPC "Memrr"
                     where mkTemplate' :: T.TemplatedOperandFn PPC "Memrr"
-                          mkTemplate' _sym locLookup = do
+                          mkTemplate' (_sym :: sym) locLookup = do
                             let offLoc = LocGPR (PPC.GPR gprOffset)
                             let baseLoc = LocGPR (PPC.GPR gprNum)
                             base <- locLookup baseLoc
                             offset <- locLookup offLoc
-                            let recover _ = do
+                            let recover :: (forall tp. S.SymExpr sym tp -> IO (S.GroundValue tp))
+                                        -> IO (A.Operand PPC "Memrr")
```

Comment:
There are two possible ways to repair this code. One is by adding a type signature like I have done here to continue allowing `recover` to be factored out with a `let`-binding. Another is to delete the `let`-binding and inline the body of `recover` like so:

```hs
                            return ( ...
                                   , T.RecoverOperandFn $ \_ -> do
                                       let gpr | gprNum /= 0 = Just (PPC.GPR gprNum)
                                               | otherwise = Nothing
                                       return $ PPC.Memrr $ PPC.MemRR gpr (PPC.GPR gprOffset)
                                   )
```

Let me know if you'd prefer the latter instead.